### DISCUSSION TITLE
How to setup a minimal Orchard Core

### DISCUSSION THREAD

**Discussion by RockNHawk** [OrchardCMS/OrchardCore#10056] [Show and tell]
if you want to setup a minimal Orchard Core, 
![image](https://user-images.githubusercontent.com/3222379/101237320-a3ae9a00-3712-11eb-8aec-15f9034a0c7b.png)

These is main project's dependies. Module project only depends on `OrchardCore.Module.Targets.csproj`.
Use it with a .recipe.json configuration for the initial Setup.

https://github.com/OrchardCMS/OrchardCore/discussions/7287

If you ConfigureServices Orchard Core with your custom config in main Application's Startup.cs , may show Unable to resolve service for type 'YesSql.ISession' while attempting to activate 'OrchardCore.Workflows.Http.Services.HttpRequestRouteActivator' error in setup
add an AddDataAccess() call in ConfigureServices will resolve this issue.
// copy & modified from AddOrchardCms
var builder = services.AddOrchardCore()
		.AddCommands()
		.AddMvc()
		.AddEmailAddressValidator()
		.AddHtmlSanitizer()
		.AddSetupFeatures("OrchardCore.Setup")
		.AddDataAccess()
		.AddDataStorage()
		.AddBackgroundService()
		//.AddTheming()
		//.AddLiquidViews()
		.AddCaching()
	;

https://github.com/OrchardCMS/OrchardCore/discussions/7288

recipe.json config：
https://github.com/OrchardCMS/OrchardCore/discussions/7287

Startup.cs config：
https://github.com/OrchardCMS/OrchardCore/discussions/7288

**Comment by ns8482e** [comment]
The easiest option is to start with SaaS recipe.

**Reply by RockNHawk** [reply]
Good idea. ha ha.

**Reply by ns8482e** [reply]
Another option is - if you don’t like to include  themes  in your orchard solution, you can reference `OrchardCore.Application.Cms.Core.Targets`

**Reply by RockNHawk** [reply]
The cms targets reference all orchard dll, sometime people want to make the application small. more useless dll also may slow down startup speed -:)

**Reply by hishamco** [reply]
Because there's no runtime dynamic compliation (Dynamic Modules), so can create your own target which has the modules you wanna use in case to minimize the number of dlls

**Reply by hishamco** [reply]
The limitation with the above approach that you will not be able you use other modules, because the are not included, so you know what I mean ;)

**Reply by RockNHawk** [reply]
Haha, thanks for your advice, user can create his own orchard target, even publish custom target to nuget :-)

**Reply by hishamco** [reply]
I suffer too when I load the full project in my laptop, but my little workaround is to disable the build for the projects that I don't need, this way I'm able to you use all the modules with no much latency ;)
